quinn/src/platform/unix.rs

```diff
@@ -151,20 +151,20 @@ impl super::UdpExt for UdpSocket {
             }
             break n;
         };
-        let ecn_bits = if let Some(cmsg) = unsafe { cmsg::Iter::new(&hdr).next() } {
-            match (cmsg.cmsg_level, cmsg.cmsg_type) {
-                // FreeBSD uses IP_RECVTOS here, and we can be liberal because cmsgs are opt-in.
-                (libc::IPPROTO_IP, libc::IP_TOS) | (libc::IPPROTO_IP, libc::IP_RECVTOS) => unsafe {
-                    cmsg::decode::<u8>(cmsg)
-                },
-                (libc::IPPROTO_IPV6, libc::IPV6_TCLASS) => unsafe {
-                    cmsg::decode::<libc::c_int>(cmsg) as u8
-                },
-                _ => 0,
-            }
-        } else {
-            0
+        let ecn_bits = match unsafe { cmsg::Iter::new(&hdr).next() } {
+            Some(cmsg @ libc::cmsghdr {
+                cmsg_level: libc::IPPROTO_IPV6, cmsg_type: libc::IPV6_TCLASS, ..
+            }) => unsafe { cmsg::decode::<libc::c_int>(cmsg) as u8 },
+            Some(cmsg @ libc::cmsghdr {
+                cmsg_level: libc::IPPROTO_IP, cmsg_type: libc::IP_TOS, ..
+            }) => unsafe { cmsg::decode::<u8>(cmsg) },
+            Some(cmsg @ libc::cmsghdr {
+                cmsg_level: libc::IPPROTO_IP, cmsg_type: libc::IP_RECVTOS, ..
+            }) => unsafe { cmsg::decode::<u8>(cmsg) },
+            Some(_) => 0,
+            None => 0,
```

Comment:
This doesn't save any lines of code even before rustfmt blows it up, and is a lot noisier; I appreciate the flattening, but I'm not sure it's a net win.

Reply:
Yeah, the old tuple match is indeed better.